src/NzbDrone.Core/History/HistoryService.cs

```diff
@@ -257,6 +258,34 @@ public void Handle(EpisodeFileDeletedEvent message)
             }
         }
 
+        public void Handle(EpisodeFileRenamedEvent message)
+        {
+            var sourcePath = message.OriginalPath;
+            var sourceRelativePath = message.Series.Path.GetRelativePath(message.OriginalPath);
+            var path = Path.Combine(message.Series.Path, message.EpisodeFile.RelativePath);
+            var relativePath = message.EpisodeFile.RelativePath;
+
+            foreach (var episode in message.EpisodeFile.Episodes.Value)
+            {
+                var history = new History
+                {
+                    EventType = HistoryEventType.EpisodeFileRenamed,
+                    Date = DateTime.UtcNow,
+                    Quality = message.EpisodeFile.Quality,
+                    SourceTitle = message.OriginalPath,
+                    SeriesId = message.EpisodeFile.SeriesId,
+                    EpisodeId = episode.Id,
+                };
+
+                history.Data.Add("SourcePath", sourcePath);
+                history.Data.Add("SourceRelativePath", sourceRelativePath);
```

Comment:
Just SourceRelativePath & RelativePath is enough, no need to add more data to the database.